system/Commands/Utilities/Optimize.php

```diff
@@ -99,32 +123,100 @@ private function removeFile(string $cache): void
         }
     }
 
-    private function enableCaching(): void
+    private function runCaching(?bool $enableConfigCache, ?bool $enableLocatorCache, ?bool $disable): void
```

Comment:
I don't understand what the `runConfig()` method dose when I see the method name.
Is `updateConfigFile()` better?

Reply:
> When I see the method name disableCaching(), I imagine the method disables caching.
But this method just returns an array.
Therefore the method name or the implementation is not good.

If `disabelCaching()` and `enableCaching()` does its job completely, I don't think `runCaching()` is needed any more. we just need a process like this.

```php
public function run(array $params)
    {
        // Parse options
        $enableConfigCache  = CLI::getOption('c');
        $enableLocatorCache = CLI::getOption('l');
        $disable            = CLI::getOption('d');

        try {
            if ($disable === true) {
                $this->disableCaching();
                $this->reinstallDevPackages();
            } else {
                $this->enableCaching(['config' => $enableConfigCache, 'locator' => $enableLocatorCache]);
                $this->removeDevPackages();
            }

            $this->clearCache();

        } catch (RuntimeException) {
            CLI::error('The "spark optimize" failed.');

            return EXIT_ERROR;
        }

        return EXIT_SUCCESS;
    }
```

Reply:
Good! That will be easier to understand.